src/codegen/codegen.cpp

```diff
@@ -127,26 +151,52 @@ llvm::Value *CodeGen::CallFunc(llvm::Value *fn,
   return GetBuilder().CreateCall(fn, args);
 }
 
-llvm::Value *CodeGen::CallPrintf(const std::string &format,
-                                 const std::vector<llvm::Value *> &args) {
+llvm::Value *CodeGen::Printf(const std::string &format,
+                             const std::vector<llvm::Value *> &args) {
   auto *printf_fn = LookupBuiltin("printf");
   if (printf_fn == nullptr) {
+#if GCC_AT_LEAST_6
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wignored-attributes"
+#endif
     printf_fn = RegisterBuiltin(
-        "printf", llvm::TypeBuilder<int(char *, ...), false>::get(GetContext()),
+        "printf", llvm::TypeBuilder<decltype(printf), false>::get(GetContext()),
         reinterpret_cast<void *>(printf));
+#if GCC_AT_LEAST_6
+#pragma GCC diagnostic pop
+#endif
   }
-  auto &ir_builder = code_context_.GetBuilder();
-  auto *format_str =
-      ir_builder.CreateGEP(ConstString(format), {Const32(0), Const32(0)});
 
   // Collect all the arguments into a vector
-  std::vector<llvm::Value *> printf_args{format_str};
+  std::vector<llvm::Value *> printf_args = {ConstString(format, "format")};
   printf_args.insert(printf_args.end(), args.begin(), args.end());
 
-  // Call the function
+  // Call printf()
   return CallFunc(printf_fn, printf_args);
 }
 
+llvm::Value *CodeGen::Memcmp(llvm::Value *ptr1, llvm::Value *ptr2,
+                             llvm::Value *len) {
+  static constexpr char kMemcmpFnName[] = "memcmp";
+  auto *memcmp_fn = LookupBuiltin(kMemcmpFnName);
+  if (memcmp_fn == nullptr) {
+#if GCC_AT_LEAST_6
```

Comment:
Again, please explain why we need this.

Reply:
I've added a short comment explaining this.

TL;DR; GCC 6+ complains when it doesn't see you using all function attributes (i.e., nothrow, throw, notnull etc.). We use `decltype` to get the signature for some functions, so we don't need the attributes. This fails compilation. We ignore this warning for this section of code.